packages/js/core/src/algorithms/parse-query.ts

```diff
@@ -46,6 +46,14 @@ export function parseQuery(
         );
       }
 
+      let alias: string | undefined = undefined;
+      if (selection.alias) {
+        alias = selection.alias.value;
+        if (invokeOptions.find((io) => io.alias === alias)) {
```

Comment:
Instead of doing this, we can change the return type of parseQuery so that it's a mapping from alias name to InvokeOptions. Here's the type:
```
interface QueryInvocations {
  [alias: string]: InvokeApiOptions;
}
```